frontend/app/components/redesign/hooks/useCopyToClipboard.ts

```diff
@@ -1,12 +1,17 @@
 import { useState, useEffect, useCallback } from 'react'
 
-export const useCopyToClipboard = (text: string) => {
+export const useCopyToClipboard = (
+  text: string,
+  onCopy?: (copied: boolean) => void
```

Comment:
`onCopy` doesn't seem the right name here. As the name suggests, `onCopy` should be fired only when it's actually copied. We're trying to emit state to parent here. Maybe renaming it to `callback` might be ok, as it won't cause this misunderstanding.